covid19_sfbayarea/news/alameda.py

```diff
@@ -56,30 +56,98 @@ def load_html(self, url: str) -> str:
+        # filter out the `None` items (those that weren't actually news).
+        parsed = (self.parse_news_item(starter, base_url)
+                  for starter in article_starts)
+        articles = [item for item in parsed if item is not None]
+
+        # Sanity check that we are parsing correctly. Because we silently
+        # filter out things that parsed incorrectly (since we only
+        # know *candidates* for news items, not things that are news items),
+        # we might not have errors if the page changes. It's more likely that
+        # we are parsing incorrectly in this case than that there was no news.
+        if len(articles) == 0:
+            raise FormatError('No news items found for Alameda County')
+
+        return articles
+
+    def parse_news_item(self, node: element.Tag, base_url: str) -> Optional[NewsItem]:
+        """
+        Parse a single news item from the page based on the first element of
+        the news item.
+
+        Parsing news items in Alameda's news page is a little rough because
+        each news item is not contained in a separate element. Instead, they
+        are a stream of elements, where each news item is separated by a double
+        `<br>` tag like so:
+
+            <strong>July 17, 2020</strong>
+            <a href="/path/to/news/item/page.html">Title of news item</a>
+            <br>
+            <strong>July 10, 2020</strong>
+            <a href="/path/to/news/item2/page.html">Title of news item 2</a>
+            <br>
+            ...
+
+        Because there's nothing to identify whether a `<strong>` is the start
+        of a news item, we first test whether its content is parseable as a
+        date. If not, it's probably not a news item and we return `None`.
+        Otherwise, we start iterating forward from the first tag until we hit
+        a `<br>`.
+        """
+        try:
+            date_string = node.get_text(strip=True)
+            date = parse_datetime(date_string)
+        except ValueError:
+            # If the first element didn't contain a date, it probably wasn't
+            # actually the start of an article. Bail out.
+            return None
+
+        # Iterate forward through the elements after the initial date to find
+        # the different parts of the news item. These generally come in one of
+        # two flavors:
+        #
+        #     <strong>July 17, 2020</strong>
+        #     <a href="/path/to/article.html">Title of news item</a>
+        #     <br>
+        #
+        # Or for items with multiple languages:
+        #
+        #     <strong>July 17, 2020</strong>
+        #     The title of the news article
+        #     <a href="/path/to/english/article.html">English</a> |
+        #     <a href="/path/to/spanish/article.html">Spanish</a> |
+        #     <br>
+        #
+        url = None
+        title = ''
+        for sibling in node.next_siblings:
+            # <br> tags signal the end of the news item.
```

Comment:
Alas, this no longer seems to be true.
![image](https://user-images.githubusercontent.com/12928553/90713431-2857d400-e25a-11ea-9f8e-411d679fb60c.png)
I think there are 2 `<br>` tags separating each news item (for now) so you could check for two in a row. Or you could check for the next `<strong>` tag. I was going to write a suggestion to break after seeing a `strong` tag but for some reason that resulted in just parsing the August news items. I confirmed that all the `strong` tags pass line 110.

Reply:
JFC. Looking at the second news item, the link’s `title` attribute implies that the `<br>` tag inside the news item is delimiting the title from a lede or subhead. So I think I’m going to take everything up to the `<br>` as the title and everything after as summary.

More fun is that the second news item has no English version, even thought the title is in English. I think I’ll treat that like non-English entries in other news feeds and not surface it. That’s definitely not ideal, but I feel like it fits best with how we’ve been treating things.

Also! Down in April, most of the entries don’t start with a `<strong>` tag at all! Since, practically speaking, we are never going to show content from that far back, I don’t think I’m going to worry about it. But still, ugh.